bundle/artifacts/artifacts.go

```diff
@@ -165,3 +165,36 @@ func getUploadBasePath(b *bundle.Bundle) (string, error) {
 
 	return path.Join(artifactPath, ".internal"), nil
 }
+
+func UploadNotebook(ctx context.Context, notebook string, b *bundle.Bundle) (string, error) {
+	raw, err := os.ReadFile(notebook)
+	if err != nil {
+		return "", fmt.Errorf("unable to read %s: %w", notebook, errors.Unwrap(err))
+	}
```

Comment:
I made a comment below about keeping this file in memory or not. If we store it on the filesystem, we should put it under some path that we overlay on top of what we sync. Then 1) these generated files can be inspected on the local filesystem, and 2) we can avoid re-synchronizing them on every deploy if they haven't changed.